### PR TITLE
[docs] C# smart driver page

### LINKED ISSUE #16205: [Docs] Add Documentation for C# smart driver
### Description

The C# smart driver has just been released. [Here](https://docs.google.com/document/d/147P8Ywziflt6Fe66G7DgtUHI91ovvDar7IqJ626uG38/edit?usp=sharing) is the documentation to be added on the [docs page](https://docs.yugabyte.com/preview/drivers-orms/csharp/postgres-npgsql/). Please let me know in case of any questions.

### PR DESCRIPTION
**Pull request by aishwarya24**
resolves #16205 
 
To clarify -
YBM steps , SSL, and driver version. 

@netlify /preview/drivers-orms/csharp/ysql/

### CODE REVIEW

File: docs/content/preview/drivers-orms/csharp/ysql.md [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+    </a>
+  </li>
+
+</ul>
+
+The [Yugabyte Npgsql smart driver](https://github.com/yugabyte/npgsql) is a distributed .NET driver for [YSQL](../../../api/ysql/) built on the [PostgreSQL Npgsql driver](https://github.com/npgsql/npgsql/tree/main/src/Npgsql), with additional [connection load balancing](../../smart-drivers/) features.
+
+{{< note title="YugabyteDB Managed" >}}
+
+To use smart driver load balancing features when connecting to clusters in YugabyteDB Managed, applications must be deployed in a VPC that has been peered with the cluster VPC. For applications that access the cluster from a non-peered network, use the upstream PostgreSQL driver instead; in this case, the cluster performs the load balancing. Applications that use smart drivers from non-peered networks fall back to the upstream driver behaviour automatically. For more information, refer to [Using smart drivers with YugabyteDB Managed](../../smart-drivers/#using-smart-drivers-with-yugabytedb-managed).
```
Comment (aishwarya24) [inline docs/content/preview/drivers-orms/csharp/ysql.md:49]: @Sfurti-yb - Can you clarify if anything else needs to be added for YBM ?

File: docs/content/preview/drivers-orms/csharp/_index.md [hunk 2]
```diff
@@ -19,6 +19,7 @@ The following projects can be used to implement C# applications using the Yugaby
 
 | Project | Documentation and Guides | Latest Driver Version | Supported YugabyteDB Version |
 | :------ | :----------------------- | :-------------------- | :--------------------------- |
+| YugabyteDB C# Driver for YSQL [Recommended] | [Documentation](ysql/) <br /> [Reference](../../reference/drivers/csharp/yb-npgsql-reference/) |
```
Comment (aishwarya24) [inline docs/content/preview/drivers-orms/csharp/_index.md:22]: @Sfurti-yb - Can I get the latest Driver version?

File: docs/content/preview/reference/drivers/csharp/yb-npgsql-reference.md [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+    connStringBuilder.RootCertificate = "/root.crt"; //Provide full path to your root CA.
+    connStringBuilder.Username = "admin";
+    connStringBuilder.Password = "xxxxxx";
+    connStringBuilder.Database = "yugabyte";
+    CRUD(connStringBuilder.ConnectionString);
+```
+
+For more information on TLS/SSL support, see [Security and Encryption](https://www.npgsql.org/doc/security.html?tabs=tabid-1) in the Npgsql documentation. -->
+
+## Compatibility matrix
```
Comment (aishwarya24) [inline docs/content/preview/reference/drivers/csharp/yb-npgsql-reference.md:196]: @Sfurti-yb - Can you mention what versions we need to mention in this Compatibility Matrix.

File: docs/content/preview/drivers-orms/csharp/ysql.md [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+NpgsqlConnection conn = new NpgsqlConnection(connStringBuilder)
+```
+
+Multiple topologies can also be passed to the `Topology Keys` property, and each of them can be given a preference value, as per the following example:
+
+```csharp
+var connStringBuilder = "Host=127.0.0.1,127.0.0.2,127.0.0.3;Port=5433;Database=yugabyte;Username=yugabyte;Password=password;Load Balance Hosts=true;Topology Keys=cloud1.region1.zone1:1,cloud2.region2.zone2:2";
+NpgsqlConnection conn = new NpgsqlConnection(connStringBuilder)
+```
+
```
Comment (ddhodge) [inline docs/content/preview/drivers-orms/csharp/ysql.md:131]: missing SSL section

File: docs/content/preview/drivers-orms/csharp/ysql.md [hunk 5]
```diff
@@ -0,0 +1,195 @@
...
+               if (conn.State != System.Data.ConnectionState.Closed)
+               {
+                   conn.Close();
+               }
+           }
+       }
+   }
+}
+```
+
```
Comment (ddhodge) [inline docs/content/preview/drivers-orms/csharp/ysql.md:181]: missing Run section. how do you run it?

File: docs/content/preview/drivers-orms/csharp/ysql.md [hunk 6]
```diff
@@ -0,0 +1,195 @@
...
+```output
+Created table Employee
+Inserted data (1, 'John', 35, 'CSharp')
+Query returned:
+Name  Age  Language
+John  35   CSharp
+```
+
+## Learn more
+
```
Comment (ddhodge) [inline docs/content/preview/drivers-orms/csharp/ysql.md:193]: See Go page for other links to add

File: docs/content/preview/drivers-orms/smart-drivers.md [hunk 7]
```diff
@@ -22,6 +22,7 @@ Yugabyte has developed the following smart drivers for YSQL, available as open s
 | [YugabyteDB PGX Driver for Go](https://github.com/yugabyte/pgx) | jackc/pgx | [Documentation](../go/yb-pgx/) |
 | [YugabyteDB Psycopg2 Driver for Python](https://github.com/yugabyte/psycopg2) | PostgreSQL psycopg2 | [Documentation](../python/yugabyte-psycopg2/) |
 | [YugabyteDB node-postgres Driver for Node.js](https://github.com/yugabyte/node-postgres) | node-postgres | [Documentation](../nodejs/yugabyte-node-driver/) |
+| [YugabyteDB Npgsql Driver for C#](https://github.com/yugabyte/npgsql) | PostgreSQL Npgsql Driver | [Documentation](../csharp/ysql/) |
```
Comment (ddhodge) [inline docs/content/preview/drivers-orms/smart-drivers.md:25]: correct github link? readme doesn't mention YB or "smart driver" at all

File: docs/content/preview/reference/drivers/ysql-client-drivers.md [hunk 8]
```diff
@@ -76,13 +76,21 @@ $ make install
 
 ## C\#
 
+### NpgsqlYugabyteDB
```
Comment (ddhodge) [inline docs/content/preview/reference/drivers/ysql-client-drivers.md:79]:
```suggestion
### Npgsql YugabyteDB Smart Driver
```

File: docs/content/preview/reference/drivers/ysql-client-drivers.md [hunk 9]
```diff
@@ -76,13 +76,21 @@ $ make install
...
+
+The YugabyteDB Npgsql Smart Driver is a distributed .NET driver for YSQL built on the PostgreSQL Npgsql driver, with features that eliminate the need for external load balancers.
```
Comment (ddhodge) [inline docs/content/preview/reference/drivers/ysql-client-drivers.md:81]:
```suggestion
The YugabyteDB Npgsql Smart Driver is a .NET driver for YSQL built on the PostgreSQL Npgsql driver, with features that eliminate the need for external load balancers.
```

File: docs/content/preview/reference/drivers/ysql-client-drivers.md [hunk 10]
```diff
@@ -160,6 +176,14 @@ To get the latest versions for projects using [Apache Maven](https://maven.apach
 
 ## Node.js
 
+### YugabyteDB node-postgres Smart Driver
+
+The YugabyteDB node-postgres Smart Driver is a distributed Node.js driver for YSQL built on the PostgreSQL node-postgres driver, with features that eliminate the need for external load balancers.
+
+For information on the YugabyteDB node-postgres Smart Driver and its load balancing features, see [YugabyteDB node-postgres Smart Driver](../../../reference/drivers/nodejs/yugabyte-pg-reference/).
+
+For building a sample Java application with the YugabyteDB JDBC driver, see [Connect an application](../../../drivers-orms/nodejs/yugabyte-node-driver/).
```
Comment (ddhodge) [inline docs/content/preview/reference/drivers/ysql-client-drivers.md:185]:
```suggestion
For building a sample node.js application with the YugabyteDB node-postgres driver, see [Connect an application](../../../drivers-orms/nodejs/yugabyte-node-driver/).
```

File: docs/content/preview/reference/drivers/ysql-client-drivers.md [hunk 11]
```diff
@@ -198,6 +222,14 @@ CentOS users can install the driver using the `sudo yum install php-pgsql` comma
 
 ## Python
 
+### YugabyteDB Psycopg2 Smart Driver
+
+The YugabyteDB Psycopg2 Smart Driver is a distributed Node.js driver for YSQL built on the PostgreSQL Psycopg2 driver, with features that eliminate the need for external load balancers.
+
+For information on the YugabyteDB Psycopg2 Smart Driver and its load balancing features, see [YugabyteDB Psycopg2 Smart Driver](../../../reference/drivers/python/yugabyte-psycopg2-reference/).
+
+For building a sample Java application with the YugabyteDB JDBC driver, see [Connect an application](../../../drivers-orms/python/yugabyte-psycopg2/).
```
Comment (ddhodge) [inline docs/content/preview/reference/drivers/ysql-client-drivers.md:231]:
```suggestion
For building a sample Python application with the YugabyteDB Psycopg2 driver, see [Connect an application](../../../drivers-orms/python/yugabyte-psycopg2/).
```

File: docs/content/preview/reference/drivers/csharp/yb-npgsql-reference.md [hunk 12]
```diff
@@ -0,0 +1,201 @@
...
+  <li >
+    <a href="../postgres-npgsql-reference/" class="nav-link">
+      <i class="icon-postgres" aria-hidden="true"></i>
+      PostgreSQL Npgsql Driver
+    </a>
+  </li>
+
+</ul>
+
+The [Yugabyte Npgsql smart driver](https://github.com/yugabyte/npgsql) is a distributed .NET driver for [YSQL](../../../api/ysql/) built on the [PostgreSQL Npgsql driver](https://github.com/npgsql/npgsql/tree/main/src/Npgsql), with additional [connection load balancing](../../smart-drivers/) features.
```
Comment (ddhodge) [inline docs/content/preview/reference/drivers/csharp/yb-npgsql-reference.md:34]: Copy Go page format for intro

File: docs/content/preview/reference/drivers/csharp/yb-npgsql-reference.md [hunk 13]
```diff
@@ -0,0 +1,201 @@
...
+
+## Fundamentals
```
Comment (ddhodge) [inline docs/content/preview/reference/drivers/csharp/yb-npgsql-reference.md:36]: Copy Go page format for driver dependency/Fundamentals sections; ie, driver dependency is its own top level section, followed by fundamentals

File: docs/content/preview/reference/drivers/csharp/yb-npgsql-reference.md [hunk 14]
```diff
@@ -0,0 +1,201 @@
...
+
+To add the NpgsqlYugabyteDB package to your project when not using an IDE, use the following `dotnet` command:
+
+```csharp
+dotnet add package NpgsqlYugabyteDB
+```
+
+or any of the other methods mentioned on the [nuget page](https://www.nuget.org/packages/Npgsql/) for NpgsqlYugabyteDB.
+
+### Connect to YugabyteDB database
```
Comment (ddhodge) [inline docs/content/preview/reference/drivers/csharp/yb-npgsql-reference.md:55]: Don't think we need to repeat all this here; see other smart driver pages for how this is arranged

**Review by ddhodge — COMMENTED** [review]
misc comments

File: docs/content/preview/drivers-orms/csharp/ysql.md (hunk 5 above)
Comment (aishwarya24) [reply docs/content/preview/drivers-orms/csharp/ysql.md:181]: Good catch !

File: docs/content/preview/drivers-orms/smart-drivers.md (hunk 7 above)
Comment (aishwarya24) [reply docs/content/preview/drivers-orms/smart-drivers.md:25]: Its a forked repo from npgsql. Not sure if they did this to some other projects too. @Sfurti-yb - I guess you must rework the readme like the Go driver : https://github.com/yugabyte/pgx

File: docs/content/preview/drivers-orms/csharp/ysql.md [hunk 15]
```diff
@@ -0,0 +1,199 @@
...
+               {
+                   conn.Close();
+               }
+           }
+       }
+   }
+}
+```
+
+To run the project `Program.cs`, choose _Run -> Start Without Debugging_.
```
Comment (ddhodge) [inline docs/content/preview/drivers-orms/csharp/ysql.md:182]: add a heading ie ## Run the application, see for example the node driver page

choose Run Start without debugging where? and what if you aren't using an IDE?

no italics. UI elements in bold. Is this referring to a specific IDE? if so, spell it out. Proper format would be something like In [name of IDE], from the **Run** menu, choose **Start Without Debugging**. If you aren't using an IDE, enter the following command: and then the command block

File: docs/content/preview/reference/drivers/csharp/yb-npgsql-reference.md [hunk 16]
```diff
@@ -0,0 +1,200 @@
...
+Yugabyte Npgsql smart driver is a .NET driver for [YSQL](../../../../api/ysql/) based on [PostgreSQL Npgsql driver](https://github.com/npgsql/npgsql/tree/main/src/Npgsql), with additional connection load balancing features.
+
+For more information on the Yugabyte Npgsql smart driver, see the following:
+
+- [YugabyteDB smart drivers for YSQL](../../../../drivers-orms/smart-drivers/)
+- [CRUD operations](../../../../drivers-orms/go/ysql)
+- [GitHub repository](https://github.com/yugabyte/npgsql)
+- [Smart Driver architecture](https://github.com/yugabyte/yugabyte-db/blob/master/architecture/design/smart-driver.md)
+
+### Download the driver dependency
```
Comment (ddhodge) [inline docs/content/preview/reference/drivers/csharp/yb-npgsql-reference.md:43]:
```suggestion
## Download the driver dependency
```

**Review by ddhodge — COMMENTED** [review]
misc comments

File: docs/content/preview/drivers-orms/csharp/_index.md (hunk 2 above)
Comment (Sfurti-yb) [reply docs/content/preview/drivers-orms/csharp/_index.md]: Yes the latest version is `8.0.0-yb-1-beta`. For YugabyteDB version - `2.8 and above`

File: docs/content/preview/drivers-orms/csharp/ysql.md (hunk 1 above)
Comment (Sfurti-yb) [reply docs/content/preview/drivers-orms/csharp/ysql.md:49]: No, this seems fine.

**Comment by ddhodge** [comment]
@Sfurti-yb What is the status of SSL - the section is currently commented out

**Review by Sfurti-yb — APPROVED** [review]
LGTM

File: docs/content/preview/drivers-orms/csharp/ysql.md [hunk 17]
```diff
@@ -130,30 +130,7 @@ NpgsqlConnection conn = new NpgsqlConnection(connStringBuilder)
...
-    connStringBuilder.Username = "admin";
-    connStringBuilder.Password = "xxxxxx";
-    connStringBuilder.Database = "yugabyte";
-    CRUD(connStringBuilder.ConnectionString);
-```
-
-[YugabyteDB Managed](https://www.yugabyte.com/managed/) clusters require SSL. Refer to [Connect applications](../../../yugabyte-cloud/cloud-connect/connect-applications/) for instructions on how to obtain the cluster connection parameters and download the CA certificate.
-
-Refer to [Configure SSL/TLS](../../../reference/drivers/csharp/yb-npgsql-reference/#configure-ssl-tls) for more information on the default and supported SSL modes, and examples for setting up your connection strings when using SSL.
+To set up the driver properties to configure the credentials and SSL certificates for connecting to your cluster, refer to [Use SSL](../postgres-npgsql/#use-ssl).
```
Comment (ddhodge) [inline docs/content/preview/drivers-orms/csharp/ysql.md:133]:
```suggestion
The YugabyteDB Npgsql smart driver support for SSL is the same as for the upstream driver. To set up the driver properties to configure the credentials and SSL certificates for connecting to your cluster, refer to [Use SSL](../postgres-npgsql/#use-ssl).
```

File: docs/content/preview/reference/drivers/csharp/yb-npgsql-reference.md [hunk 18]
```diff
@@ -141,7 +141,9 @@ while (reader.Read())
 
 ### Configure SSL/TLS
 
-The following table describes the additional parameters the YugabyteDB Npgsql smart driver requires as part of the connection string when using SSL.
+For information on using SSL/TLS for your application, refer to the .NET Npgsql driver's [Configure SSL/TLS](../postgres-npgsql-reference/#configure-ssl-tls) instructions.
```
Comment (ddhodge) [inline docs/content/preview/reference/drivers/csharp/yb-npgsql-reference.md:144]:
```suggestion
The YugabyteDB Npgsql smart driver support for SSL is the same as for the upstream driver. For information on using SSL/TLS for your application, refer to the .NET Npgsql driver's [Configure SSL/TLS](../postgres-npgsql-reference/#configure-ssl-tls) instructions.
```

**Review by ddhodge — APPROVED** [review]

**Pull request merged**